api/src/main/java/io/strimzi/api/kafka/model/status/ListenerStatus.java

```diff
@@ -28,19 +28,20 @@
         builderPackage = Constants.FABRIC8_KUBERNETES_API
 )
 @JsonInclude(JsonInclude.Include.NON_NULL)
-@JsonPropertyOrder({ "type", "addresses", "bootstrapServers", "certificates" })
+@JsonPropertyOrder({ "type", "name", "addresses", "bootstrapServers", "certificates" })
 @EqualsAndHashCode
 public class ListenerStatus implements UnknownPropertyPreserving, Serializable {
     private static final long serialVersionUID = 1L;
 
     private String type;
+    private String name;
```

Comment:
I still don't really understand exactly why we need two fields.

Reply:
I guess some discussion was happened around this -->```After some offline discussion, the type field have to be as it is today so still reporting the listener name for backward compatibility.```

Reply:
I tried to do the same using only one setter for both but I wasn't able to do so since there weremethods like `withType`  which are being used in the code, or we can say that if we don't create getter and setter for `type` field, we don't get that `withType` Method.

Reply:
I can use the `withName` method  which gets generated due to the setter of `Name` field which have same working but at the same time it will mean that the use of `type` field will be mostly over